### PR TITLE
Wip osd

### PR DESCRIPTION
**Pull request by swiftgist**
Backward compatible osd functionality, supports ceph namespace data structure, allows creation/deletion (i.e. Stage 5) of bluestore.  Initial support for bluestore + dmcrypt.

### CODE REVIEW

File: srv/salt/_modules/osd.py [hunk 1]
```diff
@@ -253,3 +292,902 @@ def restore_weight(id, **kwargs):
...
+        if self._config_version() == OSDConfig.V2:
+            return self._check_existence('journal_size', self.tli, self.device, default=self._journal_default())
+
+    def _journal_default(self):
+        """
+        """
+        if self.journal:
+            disks = __salt__['mine.get'](tgt=__grains__['id'], fun='cephdisks.list')
+            if disks:
+                for disk in self.disks[__grains__['id']]:
```
Comment (jan--f) [inline srv/salt/_modules/osd.py]: This line contains an error. Should be
```disk in disks[__grains__['id']]:```
Want me to push a fix?

**Comment by jan--f** [comment]
minor nit for usablilty: if one runs the osd module on a mon node like so:
```salt mon1 osd.is_prepared /dev/vdb``` a backtrace is returned.

Not sure if we can do something sensible about this...just cam across it.

**Comment by rjfd** [comment]
@swiftgist is it possible to provide a few instructions on how to use this new PR?

**Comment by swiftgist** [comment]
@rjfd I will, but currently this is backward compatible.  Since I need to package for the next milestone, I planned to send the instructions to a few mailing lists.  The ultimate goal is to use Jan's custom profile runner.

Take a look at the end of #62 to see the latest examples that I used.  Jan verified that these worked for him.

I have another PR incoming to deal with the removal.  That will reduce the instructions necessary for Stage 5 using dmcrypt + bluestore.

**Comment by rjfd** [comment]
@swiftgist great! thanks

**Pull request merged**